server/svix-server/src/v1/endpoints/endpoint.rs

```diff
@@ -170,6 +170,60 @@ struct RecoverIn {
     since: DateTime<Utc>,
 }
 
+#[derive(Clone, Debug, PartialEq, Validate, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct EndpointHeadersIn {
+    #[validate]
+    headers: EndpointHeaders,
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct EndpointHeadersOut {
+    headers: HashMap<String, String>,
+    sensitive: Vec<String>,
+}
+
+impl EndpointHeadersOut {
+    const SENSITIVE_HEADERS: &'static [&'static str] = &[
+        "x-auth-token",
+        "www-authenticate",
+        "authorization",
+        "proxy-authenticate",
+        "proxy-authorization",
+    ];
+
+    fn empty() -> Self {
+        Self {
+            headers: HashMap::from([]),
+            sensitive: Vec::new(),
+        }
+    }
+}
+
+impl From<EndpointHeaders> for EndpointHeadersOut {
+    fn from(hdr: EndpointHeaders) -> Self {
+        let (sens, remaining) = hdr.0.into_iter().partition(|(k, _)| {
+            Self::SENSITIVE_HEADERS
+                .iter()
+                .any(|&x| x == k.to_lowercase())
```

Comment:
See my other comment about lowercasing, that's exactly my point. You don't need it there because you have it here.

Reply:
Fixed.